Sources/QueuesFluentDriver/FluentQueue.swift

```diff
@@ -1,103 +1,97 @@
-            .where(SQLColumn("\(FieldKey.id)"), .equal, SQLBind(id.string))
+        self.sqlDb
+            .update(JobModel.sqlTable)
+            .set(\JobModel.$state, to: QueuesFluentJobState.pending)
+            .where(\JobModel.$id, .equal, id.string)
             .run()
     }
     
-    /// Currently selects the oldest job pending execution
+    // See `Queue.pop()`.
     public func pop() -> EventLoopFuture<JobIdentifier?> {
-        guard let sqlDb = db as? SQLDatabase else {
-            return self.context.eventLoop.makeFailedFuture(QueuesFluentError.databaseNotFound)
-        }
-        
-        var selectQuery = sqlDb
-            .select()
-            .column("\(FieldKey.id)")
-            .from(JobModel.schema)
-            .where(SQLColumn("\(FieldKey.state)"), .equal, SQLBind(QueuesFluentJobState.pending))
-            .where(SQLColumn("\(FieldKey.queue)"), .equal, SQLBind(self.queueName.string))
-            .where(SQLColumn("\(FieldKey.runAt)"), .lessThanOrEqual, SQLBind(Date()))
-            .orderBy("\(FieldKey.runAt)")
-            .limit(1)
-        if self.dbType != .sqlite {
-            selectQuery = selectQuery.lockingClause(SQLSkipLocked.forUpdateSkipLocked)
-        }
-        
-        var popProvider: PopQueryProtocol!
-        switch (self.dbType) {
-            case .postgresql:
-                popProvider = PostgresPop()
-            case .mysql:
-                popProvider = MySQLPop()
-            case .sqlite:
-                popProvider = SqlitePop()
-        }
-        return popProvider.pop(db: db, select: selectQuery.query).optionalMap { id in
-            return JobIdentifier(string: id)
+        self.db.eventLoop.makeFutureWithTask {
+            // TODO: Use `SQLSubquery` when it becomes available in upstream SQLKit.
+            let select = self.sqlDb
+                .select()
+                .column(\JobModel.$id)
+                .from(JobModel.self)
+                .where(\JobModel.$state, .equal, QueuesFluentJobState.pending)
+                .where(\JobModel.$queue, .equal, self.queueName.string)
+                .where(.dateValue(.function("coalesce", JobModel.sqlColumn(\.$delayUntil), SQLNow())), .lessThanOrEqual, .now())
+                .orderBy(\JobModel.$delayUntil)
+                .limit(1)
+                .lockingClause(SQLLockingClauseWithSkipLocked.updateSkippingLocked)
+            
+            if self.sqlDb.dialect.supportsReturning {
+                return try await self.sqlDb.update(JobModel.sqlTable)
+                    .set(\JobModel.$state, to: QueuesFluentJobState.processing)
+                    .set(\JobModel.$updatedAt, to: .now())
+                    .where(\JobModel.$id, .equal, .group(select.query))
+                    .returning(JobModel.sqlColumn(\.$id))
+                    .first(decodingColumn: JobModel.key(for: \.$id), as: String.self)
+                    .map(JobIdentifier.init(string:))
```

Comment:
My investigation shows that the error is thrown in here. However my limited knowledge with sql prevents me from investigating further or fixing it as I have no idea what's going on.

Reply:
Not sure if important, the postgres table `_jobs_meta is empty

![Bildschirmfoto 2024-02-14 um 23 11 05](https://github.com/vapor-community/vapor-queues-fluent-driver/assets/18739004/14118015-eb45-4c64-a8c1-6c875327da04)